File: CSharpFundamentals/CSharpOOPbasics/InterfacesAndAbstractionEx/MilitaryElitePractice/Models/Mission.cs
using MilitaryElitePractice.Contracts;
using MilitaryElitePractice.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace MilitaryElitePractice.Models
{
    public class Mission : IMission
    {
        public string CodeName { get; private set; }
        public State State { get; private set; }

        public Mission(string codeName, State state)
        {
            CodeName = codeName;
            State = state;
        }

        public void CompleteMission()
        {
            this.State = State.Finished;
        }

        public override string ToString()
        {
            return $"Code Name: {CodeName} State: {State}";
        }
    }
}

File: JavaFundamentals/JavaOOP/Interfaces-And-Abstraction-Ex/src/military_elite/Mission.java
package military_elite;

public class Mission {
    private String codeName;
    private State state;

    public Mission(String codeName, String state) {
        this.codeName = codeName;
        this.setState(state);
    }

    private void setState(String state) {
        if (state.equals("inProgress") || state.equals("finished")){
            this.state = State.valueOf(state);
        } else {
            throw new IllegalArgumentException();
        }
    }

    public State getState() {
        return this.state;
    }

    public void completeMission() {
        this.state = State.valueOf("finished");
    }

    @Override
    public String toString() {
        return String.format("Code Name: %s State: %s",
                this.codeName, this.state);
    }
}
